### PR TITLE
Add `testable` flag to mark endpoints that can be tested

### PR DESCRIPTION
**Pull request by amarthadan**

### CODE REVIEW

File: packages/node/src/handlers/test-api.ts [hunk 1]
```diff
@@ -9,17 +9,25 @@ import { WORKER_CALL_API_TIMEOUT } from '../constants';
...
-    return [new Error(`No such endpoint '${endpointName}'`), null];
+    return [new Error(`No such endpoint with ID '${endpointId}'`), null];
+  }
+
+  const endpoint = find(find(config.ois, ['title', triggerRequest.oisTitle])?.endpoints, [
+    'name',
+    triggerRequest.endpointName,
+  ]);
+  if (!endpoint || !endpoint.testable) {
+    return [new Error(`Endpoint with ID '${endpointId}' can't be tested`), null];
```
Comment (andreogle) [inline packages/node/src/handlers/test-api.ts:30]: I would say separate the endpoint doesn't exist error from the endpoint is not testable error

File: packages/node/__dev__/test-api.ts [hunk 2]
```diff
@@ -15,9 +15,9 @@ yargs(hideBin(process.argv))
     '$0',
     'Test API calls',
     {
-      'endpoint-name': {
+      'endpoint-id': {
         alias: 'e',
-        description: 'Endpoint name',
+        description: 'Endpoint id',
```
Comment (andreogle) [inline packages/node/__dev__/test-api.ts:20]: I would expand on this description. As a user, when I try to view the help/description of a command and it returns the same string as the key, I find it kinda annoying.

Maybe something like:

```suggestion
        description: 'The ID of the endpoint to be tested, which is derived from the OIS title and endpoint name',
```

File: packages/node/src/handlers/test-api.ts [hunk 3]
```diff
@@ -9,17 +9,25 @@ import { WORKER_CALL_API_TIMEOUT } from '../constants';
...
   if (!triggerRequest) {
-    return [new Error(`No such endpoint '${endpointName}'`), null];
+    return [new Error(`No such endpoint with ID '${endpointId}'`), null];
+  }
+
+  const endpoint = find(find(config.ois, ['title', triggerRequest.oisTitle])?.endpoints, [
+    'name',
+    triggerRequest.endpointName,
+  ]);
+  if (!endpoint || !endpoint.testable) {
```
Comment (andreogle) [inline packages/node/src/handlers/test-api.ts:29]: I like that this is opt-in. i.e. if you don't specify exactly `"testable": true`, then it's not testable

**Review by andreogle — APPROVED** [review]
👍🏻

File: packages/node/src/handlers/test-api.ts [hunk 4]
```diff
@@ -36,8 +44,8 @@ export async function testApi(
     clientAddress: '',
     designatedWallet: '',
     chainId: '',
-    endpointId: triggerRequest.endpointId,
-    endpointName,
+    endpointId,
+    endpointName: triggerRequest.endpointName,
```
Comment (Siegrift) [inline packages/node/src/handlers/test-api.ts:48]: Github doesn't allow me to comment on non changed lines, but can you please fix a typo on line 61?

`An unkNown error occurred`

File: packages/node/src/handlers/test-api.ts [hunk 5]
```diff
@@ -9,17 +9,25 @@ import { WORKER_CALL_API_TIMEOUT } from '../constants';
...
   const logOptions = logger.buildBaseOptions(config, { requestId: testCallId });
 
-  const triggerRequest = find(config.triggers.request, ['endpointName', endpointName]);
+  const triggerRequest = find(config.triggers.request, ['endpointId', endpointId]);
   if (!triggerRequest) {
-    return [new Error(`No such endpoint '${endpointName}'`), null];
+    return [new Error(`No such endpoint with ID '${endpointId}'`), null];
+  }
+
+  const endpoint = find(find(config.ois, ['title', triggerRequest.oisTitle])?.endpoints, [
```
Comment (Siegrift) [inline packages/node/src/handlers/test-api.ts:25]:
```suggestion
  const endpoint = find(config.ois, ['title', triggerRequest.oisTitle, "endpoints", "name", triggerRequest.endpointName])
```

doesn't work?

**Review by Siegrift — CHANGES_REQUESTED** [review]
What about the README? Shouldn't it be updated as well? 

Or it's enough to update api3 docs and tell Warren?

**Comment by amarthadan** [comment]
> What about the README? Shouldn't it be updated as well?
> 
> Or it's enough to update api3 docs and tell Warren?

I can mention the `dev:testApi` target in the README. I'll do that in the next PR with tests. Anything else should IMO go to the docs.

File: packages/node/src/handlers/test-api.ts (hunk 5 above)
Comment (Siegrift) [reply packages/node/src/handlers/test-api.ts:25]: Sry, I was confused. It won't work. But you can make the whole thing more readble by:
```
const endpoints = find(config.ois, ['title', triggerRequest.oisTitle])?.endpoints
const endpoint = find(endpoint, ["name", triggerRequest.endpointName])
```

**Review by Siegrift — APPROVED** [review]

File: packages/node/src/handlers/test-api.ts [hunk 6]
```diff
@@ -9,17 +9,27 @@ import { WORKER_CALL_API_TIMEOUT } from '../constants';
...
 
-  const triggerRequest = find(config.triggers.request, ['endpointName', endpointName]);
+  const triggerRequest = find(config.triggers.request, ['endpointId', endpointId]);
   if (!triggerRequest) {
-    return [new Error(`No such endpoint '${endpointName}'`), null];
+    return [new Error(`No such endpoint with ID '${endpointId}'`), null];
+  }
+
+  const endpoints = find(config.ois, ['title', triggerRequest.oisTitle])?.endpoints;
+  const endpoint = find(endpoints, ['name', triggerRequest.endpointName]);
```
Comment (andreogle) [inline packages/node/src/handlers/test-api.ts:26]: What happens if this `endpoints` is undefined? Maybe we need handle that error too before the second `find()`?

**Review by andreogle — CHANGES_REQUESTED** [review]

File: packages/node/src/handlers/test-api.ts (hunk 6 above)
Comment (amarthadan) [reply packages/node/src/handlers/test-api.ts:26]: We don't actually. Lodash's `find` is ok with `undefined` and it returns `undefined`, same as if the endpoint can't be found.

Comment (andreogle) [reply packages/node/src/handlers/test-api.ts:26]: oh nice 👍🏻

**Review by andreogle — APPROVED** [review]

**Pull request merged**